Doc/library/asyncio-future.rst

```diff
@@ -0,0 +1,236 @@
+      to the callback, e.g.::
+
+          # Call 'print("Future:", fut)' when "fut" is done.
+          fut.add_done_callback(
+              functools.partial(print, "Future:"))
+
+      .. versionchanged:: 3.7
+         The *context* keyword-only parameter was added.
+         See :pep:`567` for more details.
+
+   .. method:: remove_done_callback(callback)
+
+      Remove *callback* from the callbacks list.
+
+      Returns the number of callbacks removed, which is typically 1,
+      unless a callback was added more than once.
+
+   .. method:: cancel()
+
+      Cancel the Future and schedule callbacks.
+
+      If the Future is already *done* or *cancelled*, return ``False``.
+      Otherwise, change the Future's state to *cancelled*,
+      schedule the callbacks, and return ``True``.
+
+   .. method:: exception()
+
+      Return the exception that was set on this Future.
+
+      The exception (or ``None`` if no exception was set) is
+      returned only if the Future is *done*.
+
+      If the Future has been *cancelled*, this method raises a
+      :exc:`CancelledError` exception.
+
+      If the Future isn't *done* yet, this method raises an
+      :exc:`InvalidStateError` exception.
+
+   .. method:: get_loop()
+
+      Return the event loop the Future object is bound to.
+
+      .. versionadded:: 3.7
+
+   .. method:: cancelled()
+
+      Return ``True`` if the Future was *cancelled*.
+
+   .. method:: done()
+
+      Return ``True`` if the Future is *done*.
+
+      A Future is *done* if it was *cancelled* or if it has a result
+      or an exception set with :meth:`set_result` or
+      :meth:`set_exception` calls.
+
+
+In the example below, we create a Future object, schedule an
+asynchronous Task to set result for that Future, and finally, we
+wait until the Future has a result::
+
+    async def set_after(fut, delay, value):
+        # Sleep for *delay* seconds.
+        await asyncio.sleep(delay)
+
+        # Set *value* as a result of *fut* Future.
+        fut.set_result(value)
+
+    async def main():
+        # Get the current event loop.
+        loop = asyncio.get_running_loop()
+
+        # Create a new Future object.
+        fut = loop.create_future()
+
+        # Run "set_after()" coroutine in a parallel Task.
+        # We are using the low-level "loop.create_task()" API here because
+        # we already have a reference to the event loop at hand.
+        # Otherwise we could have just used "asyncio.create_task()".
+        loop.create_task(
+            set_after(fut, 1, '... world'))
+
+        print('hello ...')
+
+        # Wait until *fut* has a result (1 second) and print it.
+        print(await fut)
+
+    asyncio.run(main())
+
+
+.. note::
+
+   The Future object was designed to mimic
+   :class:`concurrent.futures.Future`.  Key differences include:
+
+   - :class:`concurrent.futures.Future` instances cannot be awaited.
+
+   - :meth:`Future.result` and :meth:`Future.exception` do not accept
```

Comment:
Be explicit about which Future result() corresponds to concurrent.futures.Future or asyncio.Future in this bullet and the next.

Reply:
very good point!